### ISSUE TITLE
Travis builds fail on python 2 with a 30% random chance.

### ISSUE THREAD

**Issue by Arusekk** [Gallopsled/pwntools#1410]
This is closely related to subprocessed tests of `pwnlib.ui`, and processes dying randomly (but only on py 2). Any ideas of how to solve this are greatly welcome.

**Comment by zer1t0** [comment]
I also have issues related to that test. To provide more information, the test which fails is in `pwnlib.ui.yesno` function, specifically:

```
>>> p = testpwnproc("print(yesno('is it ok??'))")
>>> b"is it ok" in p.recvuntil("??")
True
>>> p.sendline(b"x\nny")
>>> b"True" in p.recvall()
True
```

I suspect this is due the yesno question output ("is it ok??") it is not flushed correctly in some environments, therefore p.recvuntil() does not receive anything and the tests fails.  

Here is the error trace:

```
cument: ui
------------
**********************************************************************
File "ui.rst", line ?, in default
Failed example:
    b"is it ok" in p.recvuntil("??")
Exception raised:
    Traceback (most recent call last):
      File "/opt/python/2.7.15/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest default[1]>", line 1, in <module>
        b"is it ok" in p.recvuntil("??")
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 310, in recvuntil
        res = self.recv(timeout=self.timeout)
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 82, in recv
        return self._recv(numb, timeout) or b''
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 160, in _recv
        if not self.buffer and not self._fillbuffer(timeout):
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 131, in _fillbuffer
        data = self.recv_raw(self.buffer.get_fill_size())
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/process.py", line 711, in recv_raw
        raise EOFError
    EOFError
**********************************************************************
File "ui.rst", line ?, in default
Failed example:
    p.sendline(b"x\nny")
Exception raised:
    Traceback (most recent call last):
      File "/opt/python/2.7.15/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest default[2]>", line 1, in <module>
        p.sendline(b"x\nny")
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 775, in sendline
        self.send(line + self.newline)
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/tube.py", line 754, in send
        self.send_raw(data)
      File "/home/travis/build/Gallopsled/pwntools/pwnlib/tubes/process.py", line 727, in send_raw
        raise EOFError
    EOFError
**********************************************************************
File "ui.rst", line ?, in default
Failed example:
    b"True" in p.recvall()
Expected:
    True
Got:
    False
**********************************************************************
1 items had failures:
   3 of  25 in default
25 tests in 1 items.
22 passed and 3 failed.
***Test Failed*** 3 failures.
```

**Comment by peace-maker** [comment]
Python's `[raw_]input()` function outputs its prompt to stderr, that might confuse the doctests.

**Comment by Arusekk** [comment]
This is still true on GH Actions, most strange errors now relate to core dumps.

**Comment by zachriggle** [comment]
Is there any way to get a shell to explore the issue with core dumps?
Getting corefiles is a big PITA under Docker and friends.

**Comment by Arusekk** [comment]
Closed by #1857
